app/controllers/api/base_controller.rb

```diff
@@ -1,3 +1,27 @@
 class Api::BaseController < ApplicationController
   layout nil
-end
+
+  # POST /api/validate_email
+  def validate_email
+    email = params[:email]
+    if email_valid?(email)
+      if User.email_exists?(email)
```

Comment:
Logging email validation events without rate limiting can lead to potential abuse. Consider implementing rate limiting to prevent excessive logging and potential DoS attacks.